department-for-transport-BODS/bods
# Disruptions - Multiple disruptions on a single stop

**Multiple disruptions on a single stop**

Problem: When visualising disruptions when there is more than one on a single stop, currently the icons overlap each other

Solution: 

List multiple disruptions in a list form

Slightly move some of the disruptions so they're slightly apart 

“Spiderify” icons so we display one circle on the map, when clicked all disruption icons “spider” out from that circle (see link for examples of this: [https://bewithjonam.github.io/mapboxgl-spiderifier/docs/index.html) - number 2](https://github.com/nazka/map-gl-js-spiderfy?tab=readme-ov-file)

**Testing Instructions**
Navigate to 
<img width="694" alt="image" src="https://github.com/department-for-transport-BODS/bods/assets/59929701/fcff5c3c-5437-46ed-8119-25b98d1d03c9">

Zoom in until relatively close just enough for the cluster to not become a singular icon then click on it and watch it spiderfy and click on the icons to see if pop ups show
